qiskit_dynamics/models/generator_models.py

```diff
@@ -311,137 +276,94 @@ def frame(self) -> Frame:
+            )
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_out_of_frame_basis
+            )
 
         self._frame = Frame(frame)
 
-        self._reset_internal_ops()
-
-    @property
-    def cutoff_freq(self) -> float:
-        """Return the cutoff frequency."""
-        return self._cutoff_freq
-
-    @cutoff_freq.setter
-    def cutoff_freq(self, cutoff_freq: float):
-        """Set the cutoff frequency."""
-        if cutoff_freq != self._cutoff_freq:
-            self._cutoff_freq = cutoff_freq
-            self._reset_internal_ops()
-
-    def evaluate(self, time: float, in_frame_basis: bool = False) -> Array:
-        """Evaluate the model in array format.
+        if self._frame.frame_diag is not None:
+            self._operator_collection.apply_function_to_operators(
+                self.frame.operator_into_frame_basis
+            )
+            self._operator_collection.drift = self._operator_collection.drift - Array(
+                np.diag(self._frame.frame_diag)
+            )
 
+    def evaluate_without_state(self, time: float, in_frame_basis: Optional[bool] = False) -> Array:
+        """Evaluate the model in array format as a matrix, independent of state.
         Args:
             time: Time to evaluate the model
             in_frame_basis: Whether to evaluate in the basis in which the frame
                             operator is diagonal
-
         Returns:
-            Array: the evaluated model
-
+            Array: the evaluated model as a (n,n) matrix
         Raises:
-            QiskitError: If model cannot be evaluated.
-        """
+            QiskitError: If model cannot be evaluated."""
 
         if self._signals is None:
             raise QiskitError("""GeneratorModel cannot be evaluated without signals.""")
 
-        sig_vals = self._signals.complex_value(time)
+        sig_vals = np.real(self._signals.complex_value(time))
 
-        # evaluate the linear combination in the frame basis with cutoffs,
-        # then map into the frame
-        op_combo = self._evaluate_in_frame_basis_with_cutoffs(sig_vals)
-        return self.frame.generator_into_frame(
-            time, op_combo, operator_in_frame_basis=True, return_in_frame_basis=in_frame_basis
-        )
+        # Evaluated in frame basis, but without rotations
+        op_combo = self._operator_collection(sig_vals)
 
-    @property
-    def drift(self) -> Array:
-        """Return the part of the model with only Constant coefficients as a
-        numpy array.
-        """
-
-        # for now if the frame operator is not None raise an error
-        if self.frame.frame_operator is not None:
-            raise QiskitError(
-                """The drift is currently ill-defined if
-                               frame_operator is not None."""
-            )
+        # Apply rotations e^{-Ft}Ae^{Ft} in frame basis where F = D
+        if self.frame.frame_diag is not None:
```

Comment:
this block should just be calling a method (or method(s) if it requires multiple calls) of `Frame` directly, rather than implement it within `GeneratorModel`. If the `Frame` object is doing it less efficiently than what you've figured out with the array optimizations, then feel free to change the implementation in `Frame`!